### PR TITLE
SNEK6 AND SNEK 8 - Bugfix and add check if new Food collided with wall

### PR DESCRIPTION
**Pull request by LeitgebMathias**
SNEK6 AND SNEK 8 - Bugfix and add check if new Food would spawn on inner wall.

### CODE REVIEW

File: src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java [hunk 1]
```diff
@@ -13,27 +13,39 @@
...
      */
-    public Food(Snake snake) {
-        boolean isTargetFieldFree = true;
+    public Food(Snake snake, Wall wall) {
+        boolean isTargetFieldFree;
         int foodXCoord;
         int foodYCoord;
-        int segmentNumber = Constants.SCREEN_SIZE_MEDIUM / Constants.OBJECT_SIZE_MEDIUM;
+        int segmentNumberX = Constants.NUMBER_OF_ROWS_AND_COLS - 2 ;
+        int segmentNumberY = Constants.NUMBER_OF_ROWS_AND_COLS - 3 ;
```
Comment (sorinlzr) [inline src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java:26]: Can you please add the magic numbers in variables or add a comment to make it more clear why you substract 2 and 3 here?

File: src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java [hunk 2]
```diff
@@ -13,27 +13,39 @@
...
             int i=0;
             do {
                 if(snake.getSegments().get(i).getX() == foodXCoord &&
                         snake.getSegments().get(i).getY() == foodYCoord) isTargetFieldFree = false;
                 i++;
             } while (isTargetFieldFree && i < snake.getSegments().size());
 
+            // Check if calculated Position is inhibited by the wall
+            int j=0;
+            do {
```
Comment (sorinlzr) [inline src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java:47]: We have a loop inside a loop here, is there any way we can optimize this without using nested loops? 

Also applies for line 34

File: src/main/java/at/ac/fhcampuswien/snake/GameBoard.java [hunk 3]
```diff
@@ -114,7 +114,7 @@ private void initializeBoardObjects(Stage stage) {
         snake = new Snake(INITIAL_SIZE, INITIAL_DIRECTION);
 
         innerWall = generateRandomWall();
-        food = new Food(snake);
+        food = new Food(snake,innerWall);
```
Comment (sorinlzr) [inline src/main/java/at/ac/fhcampuswien/snake/GameBoard.java:117]: Not an issue, but before commiting you should do a code reformat (cmd+alt+L I think it's the shortcut on Mac)

**Review by sorinlzr — CHANGES_REQUESTED** [review]

File: src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java (hunk 1 above)
Comment (LeitgebMathias) [reply src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java]: Added comments with additional push.

File: src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java (hunk 2 above)
Comment (LeitgebMathias) [reply src/main/java/at/ac/fhcampuswien/snake/ingameobjects/Food.java:50]: I can't think of one right now.
I suggest we leave it that way for the beta review and if we have the time, think of a better solution before the final submission.

File: src/main/java/at/ac/fhcampuswien/snake/GameBoard.java (hunk 3 above)
Comment (LeitgebMathias) [reply src/main/java/at/ac/fhcampuswien/snake/GameBoard.java]: You are right - thank you!

**Pull request merged**